### ISSUE TITLE
Corrupt JPEG data: Premature end of data segment

### ISSUE THREAD

**Issue by mrxiaohe** [Belval/pdf2image#51]
I am trying to convert a set of PDF documents into jpegs. While converting, I occasionally get a pop up error message as shown in the screenshot below. I wonder what may have caused this and if there is a way to by pass it (since the conversion process stalls when the error dialog window is shown unless I close it).

![image](https://user-images.githubusercontent.com/3064882/53287803-59734900-3736-11e9-9777-abab2c10b60e.png)

**Comment by Belval** [comment]
Do you have a PDF I could use to reproduce the error on my side?

**Comment by Belval** [comment]
Closing for inactivity.

**Comment by bikashgupta11** [comment]
I have document for the same. However I cannot share the document. If you can suggest then I can send you logs. Basically this error comes when pdftoppm is spawned and it waits for process to complete. However it waits infinitely till OK button is pressed on error popup. I am attaching Before and After pdf
**POPUP Error**
![ImagePopUpError](https://user-images.githubusercontent.com/4414427/55227739-445e6e00-523e-11e9-9e97-d8154437a88c.PNG)

**After POPUP Error Ok pressed**
![ImagePopUpError-After](https://user-images.githubusercontent.com/4414427/55227789-635d0000-523e-11e9-8bf9-9a0e7fabdebf.PNG)

**Comment by Belval** [comment]
Hmm ok. I'll bring this up with the poppler team, maybe there is an envvars that can be set to prevent these.

**Comment by bikashgupta11** [comment]
thanks. just for information, I am using poppler for windows

**Comment by Belval** [comment]
Good news: Found the cause
Bad news: It's not overridable without recompiling libjpeg

https://github.com/libjpeg-turbo/libjpeg-turbo/blob/master/jerror.c#L104

I think I might just compile poppler on my side and distribute that instead of the alivate one.

**Comment by bhutraharish** [comment]
In my current work, I've been using the pdf2image just to get the image object of the first page of the pdf. This is how I've been using it -> convert_from_path(filePathName, dpi=400, first_page=1, last_page=1, fmt='jpg')[0]
Now, the issue in my case is that it generates a similar JPEG Error but the message is different
![ErrorCapture](https://user-images.githubusercontent.com/13233331/62592363-cebb7c00-b887-11e9-89cb-941bc9697eef.JPG)

I've checked the pdf file and there doesn't seem to be any issue with the file.
So, while this issue of windows dialog boxes for Error is still under work. Is there any way for us to just suppress the dialog box from our python script and continue with remaining tasks.

**Comment by Belval** [comment]
If there is, I haven't found it. It's packed in the library and there is no env var to deactivate it. 

On the other hand, I will have more time to spare on this project shortly so I will probably be able to rebuild said library.

**Comment by bhutraharish** [comment]
Not an issue. For the time being, I Just wrote a pywinauto script which runs concurrently and closes any such JPEG error window it finds.

**Comment by vpvsankar** [comment]
is there any alternate option to avoid this?

**Comment by Belval** [comment]
Are you still experiencing this issue? It should have been fixed by https://github.com/Belval/pdf2image/pull/195

**Comment by nasheedyasin** [comment]
> Are you still experiencing this issue? It should have been fixed by #195

I still get the "Corrupt JPEG data: Premature end of data segment" occasionally. I am using pdf2image Version: 1.16.0.

Has this PR been incorporated in the latest distributed version on PyPi?

**Comment by Belval** [comment]
Yes it should be in the latest version available on PyPI. If you are still experiencing this issue then it was not fixed. 

Unfortunately I cannot reproduce the issue so there is not much that I can do.

**Comment by nasheedyasin** [comment]
Actually updating poppler (basically using the version you've suggested in the READ.ME) worked. The issue is gone.
